src/spikeinterface/postprocessing/principal_component.py

```diff
@@ -411,10 +414,14 @@ def _fit_by_channel_local(self, n_jobs, progress_bar):
                     pca = pca_models[chan_ind]
                     pca.partial_fit(wfs[:, :, wf_ind])
             else:
-                Parallel(n_jobs=n_jobs)(
-                    delayed(partial_fit_one_channel)(pca_model_files[chan_ind], wfs[:, :, wf_ind])
-                    for wf_ind, chan_ind in enumerate(channel_inds)
-                )
+                # parallel
+                items = [(pca_model_files[chan_ind], wfs[:, :, wf_ind]) for wf_ind, chan_ind in enumerate(channel_inds)]
+                n_jobs = min(n_jobs, len(items))
+
+                with ProcessPoolExecutor(max_workers=n_jobs) as executor:
+                    results = executor.map(partial_fit_one_channel, items)
+                    for res in results:
+                        pass
```

Comment:
Maybe parralelize over units would be more efficient this would avoid starting and stoping pool many times.

Reply:
So, after some thougth, I think it's not possible to parallelize over units. In the `by_channel_local` mode, there will be 1 PCA model per channel. If we parallelize over units and have units that share some channels, their process will attempt to partial fit the same model, so we would need some synchronizer of some sort, which sounds very tricky. I think this is why parallelization was implemented at the channel level in the first place. @samuelgarcia makes sense?